### PR TITLE
chore: 🤖 use rust 1.73.0

### LINKED ISSUE #65: Rust v1.73.0

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. Closed https://github.com/rolldown-rs/rolldown/issues/65
2. Use toolchain rust 1.73.0
3. The modifications are just cargo clippy fixing .

### Test Plan

---

### CODE REVIEW

File: crates/rolldown/src/bundler/module_loader/module_loader.rs [hunk 1]
```diff
@@ -124,9 +126,7 @@ impl<'a> ModuleLoader<'a> {
   }
 
   #[allow(clippy::collection_is_never_read)]
-  async fn resolve_entries(
```
Comment (IWANABETHATGUY) [inline crates/rolldown/src/bundler/module_loader/module_loader.rs:127]: image

File: crates/rolldown/tests/esbuild/import_star/import_star_export_star_omit_ambiguous/artifacts.snap [hunk 2]
```diff
@@ -19,8 +19,8 @@ const y = 2
 
 
 var common_ns = {
-  get x() { return x },
-  get z() { return z }
+  get z() { return z },
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/import_star/import_star_export_star_omit_ambiguous/artifacts.snap:22]: Not sure if the order matters here ,it seems that the fxhash map hash order is different between 1.73.0 and 1.71.0
## 1.73.0: 
image

## 1.71.0
image

File: crates/rolldown/src/bundler/module_loader/module_loader.rs (hunk 1 above)
Comment (hyf0) [reply crates/rolldown/src/bundler/module_loader/module_loader.rs:127]: Just use `#[allow] here.

File: crates/rolldown/tests/esbuild/import_star/import_star_export_star_omit_ambiguous/artifacts.snap (hunk 2 above)
Comment (hyf0) [reply crates/rolldown/tests/esbuild/import_star/import_star_export_star_omit_ambiguous/artifacts.snap:22]: Ignore this currenly.

**Review by hyf0 — APPROVED** [review]

**Comment by hyf0** [comment]
Please use `squash and merge` to merge the PR.

**Pull request merged**